### PR TITLE
[Quickfix] Add quickfix for empty keywords #994

### PR DESCRIPTION
**Pull request by loradd**
Signed-off-by: Lorenzo Addazi <addazi.lorenzo@gmail.com>

### CODE REVIEW

**Comment by szarnekow** [comment]
Please also adjust the quickfix for spaces in keywords to not produce empty keywords.

**Comment by loradd** [comment]
> Please also adjust the quickfix for spaces in keywords to not produce empty keywords.

Should the keyword be removed or its value set using the containing rule name?

**Comment by loradd** [comment]
> Not directly related to this PR, but to the same test added before:
> `XtextGrammarQuickfixTest.fix_empty_keyword_with_spaces` fails on our milestone build. Not sure why this was not detected on the master build nor in the PR build.
> Could you try to identify what might be causing this and address this in this PR, too?
> https://ci.eclipse.org/xtext/view/Xtext-release/job/xtext-eclipse/job/milestone_2.17.0.M3/4/testReport/junit/org.eclipse.xtext.xtext.ui.editor.quickfix/XtextGrammarQuickfixTest/fix_empty_keyword_with_spaces/

Yes, I will try to identify the problem. How can I run the milestone build locally?

**Comment by kthoms** [comment]
> How can I run the milestone build locally?

- Check out branch `milestone_2.17.0.M3`
- Run the build with `mvn --batch-mode --update-snapshots -fae -Dmaven.test.failure.ignore=true -Dmaven.repo.local=.m2/repository -DJENKINS_URL=https://ci.eclipse.org/xtext -Poxygen clean install`

**Comment by kthoms** [comment]
Please note that we have something fishy going on also for Xtend. It seems related to your case and unlikely that you caused that. We will investigate this.
But if you would execute the build anyway it would be interesting if you have locally the same problem.
Also note that I set your test case to `@Ignore` on the M3 branch in order to process with the builds.

File: org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixTest.xtend [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2019 RCP Vision s.r.l. (http://www.rcp-vision.com) and others.
+ * Copyright (c) M�lardalen University (http://www.mdh.se) and others.
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixTest.xtend]: Maybe that was a misunderstanding. You cannot change the original copyright holder, e.g. Mälardalen Uni is subsummized as 'others' in this case.

Comment (loradd) [reply org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixTest.xtend]: I realized that I had inserted the wrong name and thought I could change it. I will revert the last commit, sorry.

**Comment by cdietrich** [comment]
@szarnekow @kthoms what is the current state of review?

**Comment by kthoms** [comment]
@loradd Could you rebase the branch against master, please? The diff shows unrelated changes.
The build is happy with the change, so am I.

**Comment by miklossy** [comment]
@loradd Could you please squash your commits into a single one? The [Xtext Quality Guidelines](https://github.com/eclipse/xtext/blob/master/QUALITY_GUIDELINES.md) says:

> Squash commits if they contribute to the same logical change.

**Comment by kthoms** [comment]
Let's see what the build reports, change looks good now to me.

**Comment by kthoms** [comment]
Thanks a lot, @loradd !

**Comment by loradd** [comment]
Happy to contribute!

**Pull request merged**